chain/epoch-manager/src/types.rs

```diff
@@ -166,18 +173,33 @@ impl EpochInfoAggregator {
                 .get(i)
                 .map_or::<&[(u64, u128)], _>(&[], Vec::as_slice)
                 .iter()
-                .map(|(id, _)| *id);
-            for chunk_validator_id in chunk_validators {
+                .map(|(id, _)| *id)
+                .collect_vec();
+            // The following iterates over the chunk validator assignments and gives true (if endorsement received) or false (if endorsement missed).
+            // NOTE:(#11900): If the chunk endorsements received from the chunk validators are not recorded in the block header,
+            // we use the chunk production stats as the endorsements stats, ie. if the chunk is produced then we assume that
+            // the endorsements from all the chunk validators assigned to that chunk are received.
+            let chunk_endorsements =
+                if let Some(chunk_endorsements) = block_info.chunk_endorsements() {
+                    chunk_endorsements.iter(shard_id)
+                } else {
+                    Box::new(std::iter::repeat(*mask).take(chunk_validators.len()))
+                };
```

Comment:
I personally would gate by protocol version, so that potential block info migration wouldn't impact that. But I guess if we do such migration, we'll store Option<Bitmap> or empty bitmap, so your approach makes more sense.

Reply:
Exactly, that I also plan to store the bitmap in an Option in the common BlockInfo version.